pages/es/supported-networks/cosmos.mdx

```diff
@@ -0,0 +1,191 @@
+
+There are currently three types of handlers supported for the Cosmos Hub subgraphs:
+
+- Block handlers.
+- [Event](https://docs.cosmos.network/master/core/events.html) handlers.
+- [Transaction](https://docs.cosmos.network/master/core/transactions.html) handlers.
+
+Based on the [official Cosmos documentation](https://docs.cosmos.network/):
+
+> Events are objects that contain information about the execution of the application. They are mainly used by service providers like block explorers and wallet to track the execution of various messages and index transactions.
+
+> Transactions are objects created by end-users to trigger state changes in the application.
+
+## Building a Cosmos Hub subgraph
+
+### Subgraph Dependencies
+
+[graph-cli](https://github.com/graphprotocol/graph-cli) is a CLI tool to build and deploy subgraphs, version `>=0.23.0` is required in order to work with Cosmos Hub subgraphs.
+
+[graph-ts](https://github.com/graphprotocol/graph-ts) is a library of subgraph-specific types, version `>=0.23.0` is required in order to work with Cosmos Hub subgraphs.
+
+### Subgraph Main Components
+
+There are three main key parts or files when it comes to define a subgraph:
+
+**subgraph.yaml**: a YAML file containing the subgraph manifest.
+
+**schema.graphql**: a GraphQL schema that defines what data is stored for your subgraph, and how to query it via GraphQL.
+
+**AssemblyScript Mappings**: [AssemblyScript](https://github.com/AssemblyScript/assemblyscript) code that translates from the event data to the entities defined in your schema.
+
+### Subgraph Manifest Definition
+
+The subgraph manifest (`subgraph.yaml`) identifies the data sources for the subgraph, the triggers of interest, and the functions that should be run in response to those triggers. See below for an example subgraph manifest for a Cosmos Hub subgraph:
+
+```yaml
+specVersion: 0.0.2
+description: Cosmos Subgraph Example
+schema:
+  file: ./schema.graphql # link to the schema file
+dataSources:
+  - kind: cosmos
+    network: cosmoshub-4
+    source:
+      startBlock: 0 # Required for Cosmos, set this to 0 to start indexing from chain genesis
+    mapping:
+      apiVersion: 0.0.5
+      language: wasm/assemblyscript
+      blockHandlers:
+        - handler: handleNewBlock # the function name in the mapping file
+      eventHandlers:
+        - event: rewards # the type of the event that will be handled
+          handler: handleReward # the function name in the mapping file
+      transactionHandlers:
+        - handler: handleTransaction # the function name in the mapping file
+      file: ./src/mapping.ts # link to the file with the Assemblyscript mappings
+```
+
+- Cosmos Hub subgraphs introduce a new `kind` of data source (`cosmos`).
```

Comment:
Will other Tendermint-based chains also be "cosmos", or will they have a dedicated `kind`?

Reply:
They will have a dedicated `kind`. Do you think we should rename to `cosmoshub` or something like that for better readability ?